### PR TITLE
fix: add logs and increase memory for lvgl

### PR DESCRIPTION
**Pull request by TejasvCypherock**

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-4521 BugFix: lvgl memory size increase](https://app.clickup.com/t/9002019994/CHI-4521)

File: common/startup/application_startup.c [hunk 1]
```diff
@@ -358,6 +358,7 @@ void application_init() {
         ui_rotate();
     }
     logger_init();
+    lv_log_register_print_cb((lv_log_print_g_cb_t)lv_custom_log_handler);
```
Comment (ujjwal-cyph) [inline common/startup/application_startup.c:361]: This explicit typecasting should not be needed.

File: common/logger/logger.c [hunk 2]
```diff
@@ -319,3 +318,9 @@ log_read_e_t get_log_read_status()
 {
     return sg_log_data.read_sm_e;
 }
+
+void lv_custom_log_handler(lv_log_level_t level, const char *file, int line, const char *dsc) {
```
Comment (ujjwal-cyph) [inline common/logger/logger.c:322]: the function prototype is wrong. Correction: 
```suggestion
void lv_custom_log_handler(lv_log_level_t level, const char *file, uint32_t line, const char *dsc) {
```

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]

File: common/lv_conf.h [hunk 3]
```diff
@@ -223,7 +223,7 @@ typedef void * lv_indev_drv_user_data_t;            /*Type of user data in the i
  *===============*/
 
 /*1: Enable the log module*/
-#define LV_USE_LOG      0
+#define LV_USE_LOG      1
```
Comment (ujjwal-cyph) [inline common/lv_conf.h:226]: this bumps the firmware binary size by `3256 bytes`. we might want to do this by different mechanism (directly modifying the `lv_mem_assert` macro in `lv_mem.h`.

**Review by ujjwal-cyph — APPROVED** [review]

**Pull request merged**